### PR TITLE
Improve `supervisor` specs and docs

### PR DESCRIPTION
**Pull request by Maria-12648430**
Prompted by [discussions on the Erlang Forum](https://erlangforums.com/t/args-in-gen-init-1/3169), I took a look at the `supervisor` documentation and found some ways to improve it (or so I hope). However, this PR does not really improve the issue that lead to the confusion as discussed in the linked Forum post, as the roots for this lie deeper.

I'll put some more inline comments on what I did any why.

### CODE REVIEW

**Review by elbrujohalcon — APPROVED** [review]
# 😍

**Comment by Maria-12648430** [comment]
@elbrujohalcon @vkatsuba thanks guys :hugs: I didn't expect reactions so quick TBH

File: lib/stdlib/doc/src/supervisor.xml
Comment (IngelaAndin) [reply lib/stdlib/doc/src/supervisor.xml:405]: I think it makes sense to have this type, it makes it flexible for this abstractions.

File: lib/stdlib/doc/src/supervisor.xml
Comment (IngelaAndin) [reply lib/stdlib/doc/src/supervisor.xml:446]: Yes something that we might consider is if proc_lib  (that is the main building block) should have some
register_name type that  all the behaviors and supervisors (that are implemented as gen_servers)  could refer to. But I think this is an improvement over what we have and motivate by the behaviour modules current documentation.

File: lib/stdlib/doc/src/supervisor.xml
Comment (IngelaAndin) [reply lib/stdlib/doc/src/supervisor.xml:537]: I agree that current formulation is hard to read. However clause_i will not be possible to use with the new
documentation format from #8026 so we have to come up with an alternative solution.  clause_i is sometimes convenient but also quite error prone if the code changes it implementation adds/removes clauses.  I am all for splitting it into "most supervisors"  and  "simple one for one supervisors"

File: lib/stdlib/src/supervisor.erl
Comment (IngelaAndin) [reply lib/stdlib/src/supervisor.erl:199]: Yes

**Review by IngelaAndin — CHANGES_REQUESTED** [review]

File: lib/stdlib/doc/src/supervisor.xml
Comment (Maria-12648430) [reply lib/stdlib/doc/src/supervisor.xml:537]: I guessed as much when I saw #8026... This raises the question how this PR should be handled, though. Wait for #8026 to be merged and then rework it in the "new way"?

Comment (IngelaAndin) [reply lib/stdlib/doc/src/supervisor.xml:537]: I think we do not want add new things that will need to be change once #8026 is merged.  PR-8026 will not aim to be the final version, and other changes will be needed to make everything as we like it but we need to start somewhere.
Probably for now we will have to make some solution referring to if the function is called like x then otherwise if it is called like y then ...  I have a similar problem in ssl docs but have not yet had time to make a solution.

Comment (Maria-12648430) [reply lib/stdlib/doc/src/supervisor.xml:537]: No, what I mean is, how should I proceed with this PR in general? The most sensible way seems to be to "put it on hold" for now, wait for #8026 to be merged, then sort of write it again, this part and the others, according to the new way. This is ok for me, I don't mind :wink: WDYT?

Comment (IngelaAndin) [reply lib/stdlib/doc/src/supervisor.xml:537]: Sure if you feel that is easier its planed for master anyway. I will put a stalled label on it, and remove it once #8026 is merged.

Comment (Maria-12648430) [reply lib/stdlib/doc/src/supervisor.xml:537]: I think this is the best course, yes :)

**Comment by IngelaAndin** [comment]
@Maria-12648430  Lukas has merged his doc PR  #8026

**Comment by Maria-12648430** [comment]
> @Maria-12648430 Lukas has merged his doc PR #8026

Thanks for the heads-up @IngelaAndin, I'll get back to this one in the next couple of days then :+1:

**Comment by Maria-12648430** [comment]
@IngelaAndin I changed this PR to use the new format.

The rendered documentation for the `start_child/2` function has an issue where a linebreak occurs right in the middle of `++`. I should bring that up with `ex_doc` I guess?

**Review by IngelaAndin — APPROVED** [review]

**Comment by IngelaAndin** [comment]
Thanks for the PR :)

**Pull request merged**
